src/main/java/com/mineboundteam/minebound/inventory/ArmorForgeMenu.java

```diff
@@ -158,26 +159,41 @@ public int getSize() {
 
     @Override
     public @NotNull ItemStack quickMoveStack(@NotNull Player player, int index) {
+        System.out.println(index);
         Slot slot = slots.get(index);
         if (!slot.hasItem())
             return ItemStack.EMPTY;
         ItemStack sourceStack = slot.getItem();
         ItemStack copyOfSourceStack = sourceStack.copy();
         // if in inventory, first try to move to armor slots, then into crafting slots
         if (index < 36) {
-            if (!moveItemStackTo(sourceStack, 36, 40, false)
-                    && !moveItemStackTo(sourceStack, 40, 47, false)
-                    && !moveItemStackTo(sourceStack, 0, 36, false))
+            if (sourceStack.getItem() instanceof SpellItem && slots.get(45).hasItem()) {
+                //If Spell Item and there is an Armor Item, Attempt to Move to Spell Slots
+                ItemStack armorSlotItem = slots.get(45).getItem();
+                if (armorSlotItem.getItem() instanceof MyrialArmorItem) {
+                    MyrialArmorItem armorItem = (MyrialArmorItem) armorSlotItem.getItem();
```

Comment:
You don't have to make this change, but there is a cool feature when checking `instanceOf` that allows you to bind the variable within the if statement. For example `if (armorSlotItem.getItem() instanceof MyrialArmorItem armorItem)` will do the cast and bind to `armorItem` all in that one line.